### ISSUE TITLE
macOS: "balena" cannot be opened because the developer cannot be verified

### ISSUE THREAD

**Issue by pdcastro** [balena-io/balena-cli#2244]
The standalone zip package for macOS -- one of the [advanced installation options](https://github.com/balena-io/balena-cli/blob/master/INSTALL-ADVANCED.md) -- is not currently digitally signed and notarized by Apple, having the execution blocked by the following warning popup:

<img width="532" alt="balena cannot be opened" src="https://user-images.githubusercontent.com/15091591/113786425-02bc0400-9731-11eb-960d-413ab44787b9.png">

Note that this issue does **not** apply to the graphical installer for macOS (the [recommended installation option on macOS](https://github.com/balena-io/balena-cli/blob/master/INSTALL-MAC.md)) since CLI version 12.44.4. If you are experiencing this issue with the graphical installer of CLI version 12.44.4 or later, please add a comment to issue #1479, or create a new issue.

We are investigating a solution to this issue (it appears that a zip package cannot be used to distributed software notarized by Apple). Meanwhile, please consider the workarounds below.

**Known workarounds**

A workaround is to run the following command on a terminal, **on the folder where the standalone zip package was extracted** (e.g. the `Downloads` folder), after the zip file was extracted:

```sh
$ find balena-cli -print -execdir xattr -c \{\}  \;
```

> Explanation: The `xattr -c` command has the effect of clearing all extended file attributes of the files found by the `find` command. Those extended attributes identify the files as having been downloaded from the internet and thus being untrusted, which causes macOS to raise the security popup. By the deleting the extended attributes, the security check is bypassed. This is obviously only a workaround to a packaging issue that needs fixing.

Other workarounds include:
* Use the [graphical installer](https://github.com/balena-io/balena-cli/blob/master/INSTALL-MAC.md) instead of the standalone zip package.
* Go through the tedious and convoluted series of popups described in https://github.com/balena-io/balena-cli/issues/1479#issuecomment-558148657.
